### PR TITLE
Eth/query tool rate limit. Close #2345

### LINKED ISSUE #2345: Implement rate limiting on the Query Tool's search form
For set the API Rate Limit, we need a way to match the valid request, it could be a query string variable o header variable. Right now the API Rate Limit is using the URL to match the request but the URL is going to change in production enviroment.

### PR DESCRIPTION
**Pull request by edwintorres**
Added query WAF Custom Rule Polity that matches POST request methods and applies an API Rate Limit, similar to the APIM Rate Limit #2301.

I tested this on `tts/eth` and `tts/dev` the code recreate the rule.

### CODE REVIEW

**Review by kingcomma — COMMENTED** [review]
@edwintorres Did you try running these commands a second time, after the custom rule is already in place? I'm seeing the following error when I run the commands against `tts/dev`:

```
(BadRequest) WebApplicationFirewallPolicy validation failed. More information "Custom rules rateLimitRuleByRequestMethod and rateLimitRuleByRequestMethod have duplicate priority: 1".
Code: BadRequest
Message: WebApplicationFirewallPolicy validation failed. More information "Custom rules rateLimitRuleByRequestMethod and rateLimitRuleByRequestMethod have duplicate priority: 1".
```

We've seen this happen before with other CLI commands. In those cases, we've done something like this: https://github.com/18F/piipan/blob/22137d4937723a97c7f361449656e7ba2d561580/iac/create-resources.bash#L476-L487

**Comment by edwintorres** [comment]
@kingcomma weird, yes, I ran it several times (5 or more) on `tts/eth` without error. I'll try to run it again on `tts/dev`

**Comment by kingcomma** [comment]
@edwintorres Sorry, this is my mistaken. I was running the `az network` commands on their own and not part of a full IaC run. When they are run with the full IaC, the [Front Door ARM template removes any existing custom rules](https://github.com/18F/piipan/blob/dev/iac/arm-templates/front-door-app-service.json#L82-L84) before the commands are run. That explains why I was seeing an error but not you.

**Review by kingcomma — APPROVED** [review]
👍 Just one suggestion below if it makes sense.

Thanks for the patience as I worked through how this change fit with the rest of the IaC 😅

File: iac/create-resources.bash [hunk 1]
```diff
@@ -560,6 +563,36 @@ main () {
...
+    --rate-limit-duration 5 \
+    --rate-limit-threshold 10000 --defer
+
+  echo "az network front-door waf-policy rule show"
+
+  # Show if there is any rule on the WAF policy
+  az network front-door waf-policy rule show \
+    --resource-group "$RESOURCE_GROUP" \
+    --policy-name "$QUERY_TOOL_WAF_NAME"  \
+    --name $WAF_CUSTOM_RULE_NAME
```
[the remark this answers is not in the document]
Comment (edwintorres) [reply iac/create-resources.bash]: I removed it

[the remark this answers is not in the document]
Comment (kingcomma) [reply iac/create-resources.bash]: Thanks — good to merge!

**Pull request merged**